Develop 2510 query endpoint

b97pla opened the pull request:
**What problems does this PR solve?**
This PR adds two endpoints to the REST api:
  - `view` displays the records stored in the database
  - `query` displays records from the database filtered by conditions specified in the request body

**How has the changes been tested?**
Tested on local client with dummy values

**Reasons for careful code review**
If any of the boxes below are checked, extra careful code review should be inititated.

  - [ ] This PR contains code that could remove data

Aratz commented on tests/test_models.py:
```diff
@@ -99,4 +124,104 @@ def test_failing_fetch_random_unverified_archive(self):
+        for observed_archive in observed_archives:
+            self.assertIn(observed_archive, expected_archives)
+
+    def test_query(self):
```
Not mandatory, but if you feel like it you could also add one more case in this test where you query two parameters at the same time (e.g. `description`  and `verified`, and another one where you don't specify any parameter.

b97pla replied:
yes, good idea. there is a case with two parameters (`before_date` and `after_date`) but it's not a great example since, on the test data, they give the same results whether only one is used or if both are used together..

b97pla commented on archive_db/handlers/DbHandlers.py:
```diff
@@ -189,6 +191,118 @@ def get(self):
+        body = self.decode()
+        query = self._db_query()
+
+        for parameter, condition in body.items():
+            if parameter == "path" and body[parameter]:
+                query = query.where(Archive.path.contains(body[parameter]))
+            elif parameter == "description" and body[parameter]:
+                query = query.where(Archive.description.contains(body[parameter]))
+            elif parameter == "host" and body[parameter]:
+                query = query.where(Archive.host.contains(body[parameter]))
+            elif parameter == "before_date" and body[parameter]:
+                query = query.where(
+                    Upload.timestamp <= dt.datetime.strptime(
+                        f"{body[parameter]} 23:59:59",
+                        "%Y-%m-%d %H:%M:%S"))
+            elif parameter == "after_date" and body[parameter]:
+                query = query.where(
+                    Upload.timestamp >= dt.datetime.strptime(body[parameter], "%Y-%m-%d"))
+            elif parameter == "verified" and body[parameter] in ["True", "False"]:
+                query = query.where(Verification.timestamp.is_null(body[parameter] == "False"))
+            elif parameter == "removed" and body[parameter] in ["True", "False"]:
+                query = query.where(Removal.timestamp.is_null(body[parameter] == "False"))
```
true

b97pla commented on archive_db/handlers/DbHandlers.py:
```diff
@@ -189,6 +191,118 @@ def get(self):
+        :param limit: positive integer specifying the number of rows to limit the results to
+        :return archives recorded in the database as a json object under the key "archives"
+        """
+        try:
+            limit = int(limit)
+        except ValueError:
+            limit = None
```
thanks, I guess this also ties in to the question about separate handlers 👇

b97pla commented:
thanks, great comments! I'll address your points and update the PR

b97pla commented:
@Aratz I've addressed your points. I'm intending to fix the arteria-core version and GitHub actions setup in a separate PR, so I'll skip those for now.

Aratz reviewed (approved):
Looks good, great work! Thanks for addressing my comments :)
